ui/src/Annotation.tsx

```diff
@@ -20,33 +26,71 @@ export class Annotation extends React.Component<
 > {
   constructor(props: AnnotationProps) {
     super(props);
-    this.state = { tooltipOpen: false };
+    this.state = { selected: false };
+  }
+
+  select() {
+    this.setState({ selected: true });
+    if (this.props.onSelected) {
+      this.props.onSelected();
+    }
+  }
+
+  deselect() {
+    this.setState({ selected: false });
+    if (this.props.onDeselected) {
+      this.props.onDeselected();
+    }
+  }
+
+  isSelected() {
+    return this.props.selected || this.state.selected;
+  }
```

Comment:
Could you just use `state.selected` as the source of truth, and use [React's getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) to set `state.selected` from `props.selected`?

Reply:
Good idea! Will do.

Reply:
I updated Annotation to be a controlled component so that it no longer needs to merge together props and state 🎉